File: backend/lib/models/endereco.dart
import 'cidade.dart';

class Endereco {
  final int id;
  String logradouro;
  String numero;
  String? complemento;
  String? bairro;
  String? cep;
  Cidade cidade;

  Endereco({
    required this.id,
    required this.logradouro,
    required this.numero,
    this.complemento,
    this.bairro,
    this.cep,
    required this.cidade,
  });
}

File: backend/app/Models/Endereco.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Endereco extends Model
{
    protected $table = 'endereco';
    public $timestamps = false;
    protected $fillable = [
        'logradouro', 'numero', 'complemento', 'bairro', 'cep', 'cidade_id'
    ];

    public function cidade()
    {
        return $this->belongsTo(Cidade::class, 'cidade_id');
    }

    public function pessoas()
    {
        return $this->hasMany(Pessoa::class, 'endereco_id');
    }
}
